### DISCUSSION TITLE
Minimum density and Numerical Instability

### DISCUSSION THREAD

**Discussion by MrAhriman** [firemodels/fds#13080] [Q&A]
Dear esteemed developers and experts of the FDS community,
I hope this message finds you in good spirits. As a newcomer in the field, I have been grappling with some challenges while endeavoring to construct a FDS model for tunnel fire ventilation. Specifically, I have encountered issues related to 'Minimum density' and 'ERROR: Numerical Instability'. When the fan volume rate remains constant and substantial (at 60 and 80 m³/s), the simulation proceeds smoothly, maintaining stability for at least 80 seconds without any hiccups.

However, the situation changes when I employ the RESTART function to dynamically adjust the fan volume every second before [error.txt](https://github.com/user-attachments/files/15908078/error.txt) the simulation. This approach has led to the emergence of the aforementioned 'Minimum density' and 'ERROR: Numerical Instability' issues. Accompanying this message, you will find an image depicting a density slice within the tunnel, along with the relevant FDS input file, the error log file, and a document outlining the temporal variations in fan volume (with time in the first column and corresponding fan volume in the second).

I would like to highlight that, for the fire source fuel, I have not configured it to reflect the actual scenario but have made adjustments solely to circumvent simulation errors. I am eager to receive your expert insights and guidance on how to resolve these challenges.

Thank you for your attention and assistance.

Warm regards
[Time-Volume-Temp.txt](https://github.com/user-attachments/files/15908079/Time-Volume-Temp.txt)
![最小密度错误](https://github.com/firemodels/fds/assets/97651495/7c3a933c-4272-47e7-ac7a-e08c8b5899fc)

**Comment by Er9y714** [comment]
Do you mean you create a RESTART point, change the fds file then try to restart, it gives error? 
Is the change a big step from the previous point? can you try introducing a ramp?
Did you check the number of iterations? Did you try increasing the max number of pressure iterations (e.g. to 100)?

**Reply by MrAhriman** [reply]
> Do you mean you create a RESTART point, change the fds file then try to restart, it gives error? Is the change a big step from the previous point? can you try introducing a ramp? Did you check the number of iterations? Did you try increasing the max number of pressure iterations (e.g. to 100)?

I apologize for the late response. Due to a network error, I submitted a duplicate post and subsequently closed this one.

Regarding what you mentioned, yes, I did use the restart function to adjust the wind speed. After running for a while, a numerical instability error occurred.
https://github.com/firemodels/fds/discussions/13079

**Reply by Er9y714** [reply]
No problem. 
I recommend trying RAMP and higher max number of pressure iteration.

**Reply by MrAhriman** [reply]
> No problem. I recommend trying RAMP and higher max number of pressure iteration.

With the guidance of an experienced mentor, I employed the RAMP method to adjust the wind speed. Unfortunately, despite the iterations, I encountered issues with numerical instability.

Regarding your suggestion to increase the number of pressure solver iterations, I believe you are referring to the modification of &PRES MAX_PRESSURE_ITERATIONS = 10. Is that correct?

**Reply by marcosvanella** [reply]
Can you give an idea of the total number of cells your problem has?

**Reply by rmcdermo** [reply]
Please send us the simplest input file that fails.  Thanks

**Reply by MrAhriman** [reply]
> Please send us the simplest input file that fails. Thanks

[jzw_0_5.fds.txt](https://github.com/user-attachments/files/15928555/jzw_0_5.fds.txt)
Here is the latest input file I modified, and it still reports errors. If more detailed information is required, I may need to trouble you to review it here.
https://github.com/firemodels/fds/discussions/13079

**Reply by MrAhriman** [reply]
> Can you give an idea of the total number of cells your problem has?



> Please send us the simplest input file that fails. Thanks

For the MPI computation, I divided the grid into a total of twenty groups. In my view, the error occurs in the grid group that contains the heat source. Initially, the heat source was placed between two grid groups, but later it was adjusted to be solely within one grid group. The issue of minimum density consistently arises in relation to the grid group with the heat source.
https://github.com/firemodels/fds/discussions/13079

**Reply by MrAhriman** [reply]
> No problem. I recommend trying RAMP and higher max number of pressure iteration.

As a novice, I have a question regarding the visual observation that the density appears to have decreased at the center of the flame. Upon examining the error logs, it is evident that the problematic area corresponds to the grid region in question. Could you please explain the connection between this issue and the pressure iterations? I apologize if this query causes any confusion.

**Comment by MrAhriman** [comment]
I apologize for the redundancy in creating multiple Q&A threads. You may find a more detailed description of the error in the post at https://github.com/firemodels/fds/discussions/13079.

**Comment by rmcdermo** [comment]
I thought I would have some computer resources to work this problem, but our cluster is full.  I suggest you make the following changes to your input file.
1. Do not use the wide band model for radiation until you get these other issues sorted out.
2. Add this as your PRES line, as I said, you should be using the tunnel precondittioner here.
```
&PRES SOLVER='ULMAT',VELOCITY_TOLERANCE=1.E-4,MAX_PRESSURE_ITERATIONS=20,TUNNEL_PRECONDITIONER = T/
```
If you still get an instability, then add
```
&MISC CHECK_HT=T/
```
which will reduce your time step, but will help stability.

**Reply by MrAhriman** [reply]
> I thought I would have some computer resources to work this problem, but our cluster is full. I suggest you make the following changes to your input file.
> 
> 1. Do not use the wide band model for radiation until you get these other issues sorted out.
> 2. Add this as your PRES line, as I said, you should be using the tunnel precondittioner here.
> 
> ```
> &PRES SOLVER='ULMAT',VELOCITY_TOLERANCE=1.E-4,MAX_PRESSURE_ITERATIONS=20,TUNNEL_PRECONDITIONER = T/
> ```
> 
> If you still get an instability, then add
> 
> ```
> &MISC CHECK_HT=T/
> ```
> 
> which will reduce your time step, but will help stability.

Thank you, senior, for your valuable advice and response. I will try the method you suggested. Regardless of the outcome, I am deeply grateful for the time you have taken to assist me.
